test/es-module/test-esm-export-not-found.mjs

```diff
@@ -0,0 +1,36 @@
+import '../common/index.mjs';
+import { fixturesDir } from '../common/fixtures.mjs';
+import { ok } from 'assert';
+import { spawn } from 'child_process';
+import { join } from 'path';
+import { execPath } from 'process';
+
+const importStatement =
+  'import { foo, notfound } from \'./module-named-exports.mjs\';';
+const importStatementMultiline = `import {
+  foo,
+  notfound
+} from './module-named-exports.mjs';
+`;
+
+[importStatement, importStatementMultiline].forEach((input) => {
+  const child = spawn(execPath, [
+    '--input-type=module',
+    '--eval',
+    input,
+  ], {
+    cwd: join(fixturesDir, 'es-module-loaders')
+  });
+
+  let stderr = '';
+  child.stderr.setEncoding('utf8');
+  child.stderr.on('data', (data) => {
+    stderr += data;
+  });
+  child.on('close', () => {
+    ok(stderr.toString().includes(
+      'SyntaxError: The requested module \'./module-named-exports.mjs\' ' +
+      'does not provide an export named \'notfound\''
+    ));
```

Comment:
Do we care about the full error message? I'm thinking it might be better to check for substrings like `SyntaxError:`, the containing file's name (`./module-named-exports.mjs`), and the missing named export's name (`notfound`).

```suggestion
    stderr = stderr.toString();

    ok(stderr.includes('SyntaxError:'));
    ok(stderr.includes('./module-named-exports.mjs'));
    ok(stderr.includes('notfound'));
```

(I intentionally omitted any inner quotations: we don't care if they change from single to double)

Reply:
Leaving out the quotes means that `stderr.includes('./module-named-exports.mjs')` could pass when the error actually contains `../module-named-exports.mjs` (note the two leading periods) which would be a different argument. So at least for file paths, we probably need them. And at that point, is it really worth it to split up the check to so many pieces? If the error message changes, we can just update the test; I don’t think that’s so big a deal.

The example I copied had `stderr.toString()` so I assumed there must be some cases where it wouldn’t be a string?

Reply:
Ahh, good point about the exact path.

I think yes it is worth it cuz I definitely would be annoyed to have to update the test for nothing important and wait the 4h for the tests to re-run 😝